aws/stack/register_middleware.go

```diff
@@ -0,0 +1,54 @@
+package stack
```

Comment:
Does this need to be a separate package vs `aws/middleware`?  What is the difference between the two packages?

Reply:
Can't keep the middleware helpers in `aws/middleware` as `aws/retry` and `aws/signer/v4` depend on `aws/middleware`, thus resulting in a *cyclic dependency*.

This package is supposed to be for the middleware helpers.